src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowirePassTest.php

```diff
@@ -787,4 +790,41 @@ public function testInlineServicesAreNotCandidates()
 
         $this->assertSame(array(), $container->getDefinition('autowired')->getArguments());
     }
+
+    public function testAutowireDecorator()
+    {
+        $container = new ContainerBuilder();
+        $container->register(LoggerInterface::class, NullLogger::class);
+        $container->register(Decorated::class, Decorated::class);
+        $container
+            ->register(Decorator::class, Decorator::class)
+            ->setDecoratedService(Decorated::class)
+            ->setAutowired(true)
+        ;
+
+        (new DecoratorServicePass())->process($container);
+        (new AutowirePass())->process($container);
+
+        $definition = $container->getDefinition(Decorator::class);
+        $this->assertSame(Decorator::class.'.inner', (string) $definition->getArgument(1));
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\AutowiringFailedException
+     * @expectedExceptionMessageRegExp /\$decorated1.+Symfony\\Component\\DependencyInjection\\Tests\\Compiler\\DecoratorInterface/
```

Comment:
Can you put the entire message instead?

Reply:
It will be flakier...

Reply:
What do you mean? That's what we do for most of the errors messages :)